src/frontend/window/main_window.rs

```diff
@@ -43,17 +49,31 @@ impl AppModel {
         }
     }
 
+    fn notify(&self, msg: String, urgency: Urgency) {
+        let handle = std::thread::spawn(move || {
```

Comment:
Why are you creating a new thread and WAIT for it to finish? Why not just executing the code on the main thread?

Reply:
> Why are you creating a new thread and WAIT for it to finish?

We are only waiting until the notification server received the notification.

> Why not just executing the code on the main thread?

The reason is due to the auto-exit if you select the clipboard: It's exists before it finished sending the notification to the notification-server.

Reply:
>We are only waiting until the notification server received the notification.

Still spawning a new thread is overkill.

>The reason is due to the auto-exit if you select the clipboard: It's exists before it finished sending the notification to the notification-server.

see other discussion. Creating a thread can be easily avoided.